analyses/cell-type-consensus/exploratory-notebooks/02-explore-consensus-results.Rmd

```diff
@@ -0,0 +1,291 @@
+
+```{r}
+# table with percentage of samples 
+high_tumor_df |> 
+  dplyr::select(project_label, percentage_unknown) |> 
+  dplyr::arrange(desc(percentage_unknown))
+
+```
+
+It looks like all projects do have cell types identified that are not "Unknown". 
+However, `SCPCP000011` (retinoblastoma), has a fairly high percentage of samples without any consensus labels. 
+
+## Number of cell types observed
+
+Below we look at the number of cell types observed in each project for all samples. 
+This does not include cells labeled as "Unknown". 
+
+```{r, fig.height=10}
+num_celltypes_df <- all_results_df |> 
+  # add a new line for facet labels 
+  dplyr::mutate(facet_label = glue::glue("{project_id}\n{diagnosis}")) |>
+  # remove unknown as a cell type 
+  dplyr::filter(consensus_annotation != "Unknown") |> 
+  dplyr::select(facet_label, library_id, num_celltypes) |> 
+  unique()
+
+ggplot(num_celltypes_df, aes(x = num_celltypes)) +
+  geom_histogram(binwidth = 1, center = 0) +
+  facet_wrap(vars(facet_label), 
+             ncol = 3) +
+  labs(
+    x = "Number of cell types"
+  ) +
+  theme_bw()
+```
+
+## Distribution of consensus cell types 
+
+Now we look at the distribution of the cell types in each sample. 
+For these plots, we will pull out the top 9 cell types for each project. 
+All other cells will be labeled with "All remaining cell types". 
+
+The top cell types are determined by counting how many libraries each cell type is found in within a project and taking the most frequent types. 
+
+```{r}
+plot_df <- all_results_df |> 
+    dplyr::group_by(project_id) |> 
+    dplyr::mutate(
+      # get most frequently observed cell types across libraries in that project 
+      top_celltypes = forcats::fct_lump_n(consensus_annotation, 9, other_level = "All remaining cell types", ties.method = "first") |> 
+        # sort by frequency 
+        forcats::fct_infreq() |> 
+        # make sure all remaining and unknown are last, use this to assign colors in specific order
+        forcats::fct_relevel("All remaining cell types", "Unknown", after = Inf)
+    )
+
+# get all unique cell types ordered by frequency 
+unique_celltypes <- plot_df |> 
+  dplyr::filter(!top_celltypes %in% c("All remaining cell types", "Unknown")) |> 
+  dplyr::pull(top_celltypes) |> 
+  unique() |>
+  sort() |> 
+  as.character()
+
+# get color palette
+colors <- c(
+  palette.colors(palette = "alphabet"),
+  "black", # 1 extra since alphabet is 26 and we have 27, this will be plasma cell which shows up once 
+  "grey60", 
+  "grey95"
+)
+names(colors) <- c(unique_celltypes, "All remaining cell types", "Unknown")
+```
+
+
+```{r, fig.height=60, fig.width=10}
+project_labels <- unique(all_results_df$project_label)
+
+# stacked bar chart showing the distribution of the top 9 cell types for each project, including Unknown
+project_labels |> 
+  purrr::map(\(label){
+    
+    project_df <- plot_df |> 
+      dplyr::filter(project_label == {{label}}) |> 
```

Comment:
I don't think you answered whether the `{{` was needed, but I don't think it is?
```suggestion
      dplyr::filter(project_label == label) |> 
```